### PR TITLE
Adjusted distributed workloads UI metrics test keyword for post upgra…

### PR DESCRIPTION
**Pull request by abhijeet-dhumal**
[RHOAIENG-12445](https://issues.redhat.com/browse/RHOAIENG-12445)
Adjusted distributed workloads UI metrics test keyword for post upgrade test to verify workload created date component value is correctly formatted and to verify it's value to be less than or equal the current date

Pre-upgrade test : [/job/devops/job/rhoai-test-flow/405](https://%3Copendatascience-jebkins%3E/job/devops/job/rhoai-test-flow/405)
Post-upgrade test : [/job/devops/job/rhoai-test-flow/406](https://%3Copendatascience-jebkins%3E/job/devops/job/rhoai-test-flow/406)

image

### CODE REVIEW

File: ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource [hunk 1]
```diff
@@ -333,6 +333,23 @@ Check Distributed Worklaod Status Overview
...
+    [Documentation]    checks Distributed Worklaod Status Overview displaying correctly
+    [Arguments]    ${job_name}    ${job_status}    ${job_status_message}
+    Click Button    ${WORKLOAD_STATUS_TAB_XP}
+    Wait Until Element Is Visible    xpath=//div[text()="Distributed Workloads"]    timeout=20
+    Check Expected String Equals    //*[@id="status-overview-ChartLabel-title"]    1Distributed Workloads
+    Page Should Contain    text=${JOB_STATUS}: 1
+    Check Expected String Contains    //td[@data-label="Name"]    ${job_name}
+    Check Expected String Equals    //td[@data-label="Priority"]    0
+    ${current_date} =    Get Current Date    result_format=%m/%d/%Y
+    ${ui_fetched_datetime} =    Get Text  xpath=//td[@data-label="Created"]
```
Comment (ChughShilpa) [inline ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource:346]: Create a separate method for L346-L349 as like we have https://github.com/red-hat-data-services/ods-ci/blob/8b4077e8d5e10591546ead692763e1ce29d9a4f7/ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource#L332

**Review by ChughShilpa — CHANGES_REQUESTED** [review]
Try to avoid code duplication between [existing method](https://github.com/red-hat-data-services/ods-ci/blob/8b4077e8d5e10591546ead692763e1ce29d9a4f7/ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource#L332) and [new method  ](https://github.com/red-hat-data-services/ods-ci/blob/8b4077e8d5e10591546ead692763e1ce29d9a4f7/ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource#L336)

File: ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource (hunk 1 above)
Comment (ChughShilpa) [inline ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource:346]: Rename `${ui_fetched_datetime}` it to get_created_datetime

**Review by jiripetrlik — APPROVED** [review]

File: ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource (hunk 1 above)
Comment (abhijeet-dhumal) [reply ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource]: Hey @ChughShilpa , Thanks for your review!
Yes, you're right! The repetition is not needed in this case so I 'm nowe utilising previous keyword with some tweaks now it should work.
I have created separate keyword for above mentioned lines.. Please review!

File: ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource [hunk 2]
```diff
@@ -329,7 +335,9 @@ Check Distributed Worklaod Status Overview
     Check Expected String Contains    //td[@data-label="Name"]    ${job_name}
     Check Expected String Equals    //td[@data-label="Priority"]    0
     ${get_date} =    Get Current Date    result_format=%-m/%-d/%Y
-    Check Expected String Contains    //td[@data-label="Created"]    ${get_date}
+    ${get_created_datetime} =    Get Text  xpath=//td[@data-label="Created"]
+    ${ui_parsed_date} =    Evaluate    re.search(r'[^,]+','''${get_created_datetime}''').group(0)
```
Comment (ChughShilpa) [inline ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource:339]: Please be consistent with the variable name.
Suggestion: Rename `${ui_parsed_date}` it to `get_created_date`

**Comment by ChughShilpa** [comment]
@abhijeet-dhumal Could you please run the upgrade tests and UI test in jenkins and share the result

File: ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource [hunk 3]
```diff
@@ -267,6 +267,12 @@ Check Distributed Workload Resource Metrics Status
     Check Expected String Contains    //td[@data-label="Name"]    ${job_name}
     Check Expected String Equals    //td[@data-label="Status"]//span[@class="pf-v5-c-label__text"]    ${job_status}
 
+Compare Workload Created Date With Current Date
+    [Documentation]    Check Workload Created Date To Be Less Than Equal To Current Date
+    [Arguments]    ${workload_created_date}    ${current_date}
+    ${check_workload_created_date}    Evaluate    datetime.datetime.strptime('${workload_created_date}','%m/%d/%Y') <= datetime.datetime.strptime('${current_date}','%m/%d/%Y')
+    Should Be True    ${check_workload_created_date}
```
Comment (sutaakar) [inline ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource:274]: Can you please improve assertion handling error message?
Currently the line:
```
Compare Workload Created Date With Current Date    9/20/2024    9/19/2024
```
provides error message:
```
'False' should be true.
```
which doesn't provide much context.

Comment (abhijeet-dhumal) [reply ods_ci/tests/Resources/Page/DistributedWorkloads/WorkloadMetricsUI.resource]: I have added meaningful error message for failed case https://github.com/red-hat-data-services/ods-ci/pull/1792/commits/8fa11ff2a6b9cd742f941cd63a2c076224490cfd
Thanks @sutaakar for reviewing this PR !

**Comment by abhijeet-dhumal** [comment]
> @abhijeet-dhumal Could you please run the upgrade tests and UI test in jenkins and share the result

@ChughShilpa 
I have tested this PR using pre/post-upgrade tests : 
Pre-upgrade test : [/job/devops/job/rhoai-test-flow/405](https://<opendatascience-jebkins>/job/devops/job/rhoai-test-flow/405)
Post-upgrade test : [/job/devops/job/rhoai-test-flow/406](https://<opendatascience-jebkins>/job/devops/job/rhoai-test-flow/406)

**Review by sutaakar — APPROVED** [review]

**Review by ChughShilpa — APPROVED** [review]

**Pull request merged**